### PR TITLE
convert all `stop()` and `rlang::abort()` commands to `cli::cli_abort()`

### LINKED ISSUE #132: align utility function prefixes
We have a variety of `validate_*()`, `abort_*()`, and `stop_if_not_*()` functions here (I take significant responsibility for that, though the misalignment was mostly created when importing the functions from pacta.multi.loanbook.analysis and pacta.multi.loanbook.plot).

Additionally, through my experience building pacta.data.validation, I came to the conclusion of using `assert_*()` and `is_*()` as [prefixes for such functions](https://github.com/RMI-PACTA/pacta.data.validation/issues/74), for the purpose of keeping the prefixes reasonably short while also clearly identifying which functions throw an error versus those that return `TRUE` or `FALSE`.

@jacobvjk @jdhoffa I'd like to decide which style we prefer here and implement that universally in this package.

- related #227
- related #228
- related #229

### LINKED ISSUE #228: harmonize abort commmands to use `cli::cli_abort()`
- related #229

We currently use a combination of `cli::cli_abort()`, `rlang::abort()`, and `stop()`

### PR DESCRIPTION
**Pull request by cjyetman**
- closes #228

### CODE REVIEW

File: R/aggregate_alignment_loanbook_exposure.R [hunk 1]
```diff
@@ -43,7 +43,10 @@ aggregate_alignment_loanbook_exposure <- function(data,
 
   if (!is.null(.by)) {
     if (!inherits(.by, "character")) {
-      stop(glue::glue("`.by` must be a character vector. Your input is {class(.by)}."))
+      cli::cli_abort(c(
+        x = "{.arg .by} must a {.cls character} vector",
```
Comment (jacobvjk) [inline R/aggregate_alignment_loanbook_exposure.R]:
```suggestion
        x = "{.arg .by} must be a {.cls character} vector",
```

File: R/plot_scatter.R [hunk 2]
```diff
@@ -211,23 +211,32 @@ plot_scatter <- function(data,
 
 
 check_plot_scatter <- function(data, alignment_limit, cap_outliers, floor_outliers) {
-  abort_if_missing_names(data, c(
+  assert_missing_names(data, c(
```
Comment (jacobvjk) [inline R/plot_scatter.R]: shouldn't the naming be something along the lines of `assert_no_missing_names()`?

File: R/prep_sankey.R [hunk 3]
```diff
@@ -92,40 +92,39 @@ check_prep_sankey <- function(data_alignment,
...
-      x = glue::glue("You provided year = {year}.")
+    cli::cli_abort(c(
+      x = "{.arg year} value not found in {.var data_alignment} dataset",
+      i = "{cli::qty(length(unique(data_alignment$year)))}year{?s} in
+           {.var data_alignment} {?is/are}: {.val {unique(data_alignment$year)}}",
+      i = "the value{?s} in {.arg year} {?is/are}: {.val {unique(year)}}"
     ))
   }
-  abort_if_middle_node_column_not_found(data_alignment, middle_node, env = list(data = substitute(data_alignment)))
+  assert_middle_node_column_not_found(data_alignment, middle_node, env = list(data = substitute(data_alignment)))
```
Comment (jacobvjk) [inline R/prep_sankey.R]: shouldn't the naming be something like `assert_middle_node_column_exists()`?

File: R/prep_scatter.R [hunk 4]
```diff
@@ -108,8 +108,8 @@ check_prep_scatter <- function(data,
       value_col
     )
   )
-  abort_if_unknown_values(sector, data, "sector")
-  abort_if_unknown_values(region, data, "region")
-  abort_if_unknown_values(year, data, "year")
-  abort_if_unknown_values(groups_to_plot, data, group_var)
+  assert_unknown_values(sector, data, "sector")
```
Comment (jacobvjk) [inline R/prep_scatter.R]: `assert_no_unknown_values()`?

**Review by jacobvjk — CHANGES_REQUESTED** [review]
one tiny correction and a few questions on `assert_` naming

File: R/plot_scatter.R (hunk 2 above)
Comment (cjyetman) [reply R/plot_scatter.R]: ah crap, you're right, got sloppy towards the end. That's why I prefer the assertive versions of these names e.g. `assert_*()`... less confusing than the double negatives.

File: R/prep_sankey.R (hunk 3 above)
Comment (cjyetman) [reply R/prep_sankey.R]: done

File: R/prep_scatter.R (hunk 4 above)
Comment (cjyetman) [reply R/prep_scatter.R]: done

File: R/plot_scatter.R (hunk 2 above)
Comment (cjyetman) [reply R/plot_scatter.R]: done

**Review by jacobvjk — APPROVED** [review]

**Pull request merged**